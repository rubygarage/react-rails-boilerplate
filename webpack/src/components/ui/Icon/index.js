import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

function Icon({ type, className, onClick }) {
  return <i className={classNames('icon', type, className)} onClick={onClick} />
}

Icon.propTypes = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func
}

export default Icon
