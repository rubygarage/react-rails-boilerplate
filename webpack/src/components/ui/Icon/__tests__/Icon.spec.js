import React from 'react'
import { shallow } from 'enzyme'
import Icon from 'components/ui/Icon'

describe('<Icon />', () => {
  const props = {
    type: 'arrows',
    className: 'TestClass',
    onClick: jest.fn()
  }

  it('renders with empty props', () => {
    const component = shallow(
      <Icon />
    )

    expect(component).toMatchSnapshot()
  })

  it('renders with props', () => {
    const component = shallow(
      <Icon {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})
