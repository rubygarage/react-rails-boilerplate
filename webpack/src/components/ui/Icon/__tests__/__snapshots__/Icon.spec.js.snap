// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders with empty props 1`] = `
<i
  className="icon arrows"
/>
`;

exports[`<Icon /> renders with props 1`] = `
<i
  className="icon arrows TestClass"
  onClick={[Function]}
/>
`;
